wpilib/wpilib/sendablechooser.py

```diff
@@ -88,19 +82,13 @@ def getSelected(self):
 
         :returns: the object associated with the selected option
         """
-        selected = None
+        selected = self.defaultChoice
         if self.tableSelected is not None:
             selected = self.tableSelected.getString(None)
         return self.map.get(selected, self.map[self.defaultChoice])
 
-    def getSmartDashboardType(self):
-        return "String Chooser"
-
-    def initTable(self, table):
-        if table is not None:
-            self.tableDefault = table.getEntry(self.DEFAULT)
-            self.tableSelected = table.getEntry(self.SELECTED)
-            self.tableOptions = table.getEntry(self.OPTIONS)
-            self.tableOptions.setStringArray(self.map.keys())
-            if self.defaultChoice is not None:
-                self.tableDefault.setString(self.defaultChoice)
+    def initSendable(self, builder):
+        builder.setSmartDashboardType("String Chooser")
+        builder.addStringProperty(self.DEFAULT, lambda: self.defaultChoice, None)
+        builder.addStringArrayProperty(self.OPTIONS, lambda: self.map.keys(), None)
```

Comment:
`self.map.keys` instead of `lambda: self.map.keys()`? `self.map` should always be the same object.

Reply:
This comment was never addressed, BTW. Doesn't really matter though.

Reply:
i think it's more robust the way it is